### PR TITLE
PEREX-1295 feat: add better logging to personas SMS action

### PR DESCRIPTION
**Pull request by cfree**
Adds better logging to personas SMS action to match the logging in the personas email action
[PEREX-1295](https://segment.atlassian.net/browse/PEREX-1295)

## Testing

- [ ] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [ ] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [x] [Segmenters] Tested in the staging environment

### CODE REVIEW

**Review by mugelstad — APPROVED** [review]
Looks good to me. Seeing "Unexpected any. Specify a different type" warnings from the file changes. Since this can be safely ignored, I am approving.

**Review by dlasky — APPROVED** [review]
Probably just error complaining. You can always catch(err) {
const error as Error

File: packages/destination-actions/src/destinations/personas-messaging-sendgrid/sendEmail/index.ts [hunk 1]
```diff
@@ -51,7 +51,7 @@ const fetchProfileTraits = async (
     const body = await response.json()
     return body.traits
   } catch (error) {
-    throw new IntegrationError('Unable to get profile traits', 'Trait fetch failure', 500)
+    throw new IntegrationError('Unable to get profile traits for email message', 'Email trait fetch failure', 500)
```
Comment (pmunin) [inline packages/destination-actions/src/destinations/personas-messaging-sendgrid/sendEmail/index.ts:54]: maybe good idea to include error.toString() or error.message into the IntegrationError constructor parameters? same in other files below

**Review by marinhero — APPROVED** [review]

File: packages/destination-actions/src/destinations/personas-messaging-sendgrid/sendEmail/index.ts (hunk 1 above)
Comment (cfree) [reply packages/destination-actions/src/destinations/personas-messaging-sendgrid/sendEmail/index.ts:54]: Good call, definitely useful for when logging is ready. This is customer-facing, so we probably don't want to get too technical.

**Pull request merged**